SoomlaiOSProfile/SoomlaProfile.m

```diff
@@ -100,9 +104,14 @@ - (void)loginWithProvider:(Provider)provider andPayload:(NSString *)payload andR
         [authController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
     }
     @catch (NSException *exception) {
-
-        // TODO: implement logic like in java that will raise the exception. Currently not raised
-        [socialController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
+        @try {
+            // TODO: implement logic like in java that will raise the exception. Currently not raised
+            [socialController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
+        }
+        @catch (NSException *exception) {
+            // TODO: implement logic like in java that will raise the exception. Currently not raised
+            [gameServicesController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
```

Comment:
@borisspektor @vedi, what about such thing?
`- (void)loginWithProvider:(Provider)provider andPayload:(NSString *)payload andReward:(Reward *)reward {
    for (AuthController *controller in @[authController, socialController, gameServicesController]) {
        @try {
            [controller loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
            break;
        }
        @catch (NSException *exception) {

        }

    }
}`